### PR TITLE
meterfs: readRecord rewrite - fix CRC error on read with offset

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-178

## Motivation and Context

https://github.com/phoenix-rtos/phoenix-rtos-project/issues/375

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (stm32l4).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/105 after this one).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
@xvuko Your test seem to pass

File: meterfs/meterfs.c [hunk 1]
```diff
@@ -843,90 +843,70 @@ static int meterfs_readRecord(file_t *f, void *buff, size_t bufflen, unsigned in
...
 
 	/* Calculate record position in a storage */
 	pos = (f->firstoff / (f->header.recordsz + sizeof(entry_t))) + idx;
 	pos = pos % ((f->header.sectorcnt * ctx->sectorsz) / (f->header.recordsz + sizeof(entry_t)));
 	addr = pos * (f->header.recordsz + sizeof(entry_t)) + f->header.sector * ctx->sectorsz;
 
-	if (bufflen > f->header.recordsz - offset)
+	if (bufflen > f->header.recordsz - offset) {
 		bufflen = f->header.recordsz - offset;
+	}
```
Comment (xvuko) [inline meterfs/meterfs.c:865]: `offset > f->header.recordsz` handling should be added here

Comment (agkaminski) [reply meterfs/meterfs.c:865]: It's handled in readFile function (only one to call readRecord (https://github.com/phoenix-rtos/phoenix-rtos-filesystems/blob/3574e7bbd1fcfc6d7610c07b54a263237d498771/meterfs/meterfs.c#L1173)

**Review by anglov — APPROVED** [review]
Code looks good to me butI didn't tested it. 
I know that @xvuko run some additional test so please confirm that it's good.

File: meterfs/meterfs.c (hunk 1 above)
Comment (xvuko) [reply meterfs/meterfs.c:865]: Yes it is.

**Review by xvuko — APPROVED** [review]

**Pull request merged**